### PR TITLE
Add to developer guide on QuizQuestionsManager and quiz logic

### PR DESCRIPTION
**Pull request by Zhi-You**
For #61 .

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -16,6 +16,33 @@ When the user launches the app, the main program will initialize a `TopicsStorag
 
 ![TopicsStorage load](./images/TopicsStorage_load.png)
 
+### Design of the Quiz system
+
+![QuizQuestionsManager_Class_Diagram](./images/QuizQuestionsManager.png)
+
+The Class Diagram given above explains the high-level design of the Quiz system in E-Duke-8. Given below is a quick overview of each component.
```
Comment (anqi20) [inline docs/DeveloperGuide.md]: Would it be better if we have a few words to introduce the diagram first?

**Review by anqi20 — APPROVED** [review]
Everything looks good!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Zhi-You) [reply docs/DeveloperGuide.md]: Alright! Thanks! I will change accordingly!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -61,6 +61,36 @@ When the user launches the app, the main program will initialize a `TopicsStorag
...
+The Sequence Diagram below shows how `QuizQuestionsManager` achieves this for the scenario where the user indicates that he wants to attempt 5 questions from the topic on OOP, which translates to the `setQuizQuestions(5, questionsInTopic)` call:
+
+
+![QuizQuestionsManager_setQuizQuestions](./images/QuizQuestionsManager_setQuizQuestions.png)
+
+`nextInt(5)` is a method call to an object of the `Random` class. It returns a random integer between 0 (inclusive) and the number passed in as argument, 5 in this scenario, exclusive. 
+
+To determine if randomQuestionIndex is not selected before, an integer ArrayList is initialized to record all the selected integers. By checking against this collection of integers, it can be determined if a currently selected integer is repeated or not, and if it is, no question will be added for that iteration of the loop. 
+
+An ArrayList of `Question` objects stores all the selected questions meant for the quiz.
```
Comment (josephhhhhhhhh) [inline docs/DeveloperGuide.md:91]: Looks really good, the explanation is methodical, while using concise language. Nice job!

**Review by josephhhhhhhhh — APPROVED** [review]
Everything looks good here, so I will be merging this in.

**Pull request merged**